### ISSUE TITLE
Circular structure error in 0.2.0

### ISSUE THREAD

**Issue by mintyfloss** [FormidableLabs/next-urql#5]
I just upgraded from 0.1.1 to 0.2.0 and now I'm seeing the error below

```
Circular structure in "getInitialProps" result of page "/mobile-store". https://err.sh/zeit/next.js/circular-structure
```

**Comment by parkerziegler** [comment]
Interesting. I haven't encountered this before, but I think it could be caused if you're attempting to pass the entire `req` object off of `ctx` down in `getInitialProps`. Read here for more details: https://github.com/zeit/next.js/issues/7371 Can you see if the solutions offered in that thread help fix your problems? Also, if you're able to please post your `withUrqlClient` setup, that may help us narrow down the issue. A minimal reproduction on CodeSandbox is also super helpful!

**Comment by kriscarle** [comment]
I just hit this as well. The issue seems to be here https://github.com/FormidableLabs/next-urql/blob/c207c9b317be42dcce1e8d3b3097f4ea03bd5539/src/with-urql-client.tsx#L111  The entire ctx is being passed down into props, including `ctx.req`. In that nextjs issue they are saying that won't work because it isn't always serializable? 

Unfortunately in my case access to `ctx.req` was exactly why I was trying out v0.2.0 to see if I could switch from my custom implementation :)

Maybe we just need to hardwire it for something like `req.urql` then pass that down to HOC function instead of the full context object? Then it would be up to us to make sure our server side app is putting whatever we need into `req.urql` In my case I would add a quick middleware function to Express to set `req.urql = { token: req.session.user.token }` after the session is loaded.

It could also be a security concern to serialize the entire context including `ctx.req` and send it to the client? In some apps there may be sensitive user session data in there?

**Comment by suevalov** [comment]
I'm trying to run `example` and getting the same error.

**Comment by parkerziegler** [comment]
@kriscarle thanks for the investigative work. This HoC was created mostly from the `urql` Next example over in the `urql` repo: https://github.com/FormidableLabs/urql/blob/master/examples/3-ssr-with-nextjs/src/with-urql-client.js What I'm realizing about that implementation is that it both implements `getInitialProps` _and_ has a fallback calling `initUrqlClient` call in the HoC itself. I'm wondering if we might be able to remove that fallback `initUrqlClient` call, in which case we wouldn't need to pass `ctx` down. Let me check in on that implementation and see if it can help us out. Sorry for the issues here, we'll get this sorted – a lot of us at Formidable are just fully booked on other work and doing this on evenings and weekends.

**Comment by parkerziegler** [comment]
👋 Hey folks, just released this in [v0.2.1](https://github.com/FormidableLabs/next-urql/releases). Mind giving it a try and letting me know if things are fixed on your ends? I tested this much more aggressively this time around, so I think we should be all set.
